portal-impl/src/com/liferay/portal/dao/orm/common/SQLTransformer.java

```diff
@@ -114,10 +89,9 @@ private String _transformFromJPQLToHQL(String sql) {
 		return newSQL;
 	}
 
-	private static final SQLTransformer _instance = new SQLTransformer();
-
-	private com.liferay.portal.dao.sql.transformer.SQLTransformer
-		_sqlTransformer;
-	private Map<String, String> _transformedSqls;
+	private static com.liferay.portal.dao.sql.transformer.SQLTransformer
+		_sqlTransformer = SQLTransformerFactory.getSQLTransformer(
+			DBManagerUtil.getDB());
```

Comment:
@marianoalvarosaiz, please remove the static field _sqlTransformer initial value. The value will be overridden by reloadSQLTransformer(). By checking reloadSQLTransformer() usages, the field _sqlTransformer value should be assigned in the reloadSQLTransformer() and this will keep consistent with the original SQLTransformer did.

Reply:
@yuhai I had that feeling also, but after giving the code a second check I found that that's not correct. In `DBInitUtil._init` if `_checkDefaultRelease` returns `false` the following alter table will execute different `runSQL` statements calling the transform method before the `reloadSQLTransformer` gets called.

In fact previous `SQLTransformer` logic ensured that `com.liferay.portal.dao.sql.transformer.SQLTransformer` was created before any transform method because the `reloadSQLTransformer` was called in the private instance constructor.

Reply:
@marianoalvarosaiz, got it. In my opinion, the `_sqlTransformer` was assigned value twice is unnecessary. Or we may only replace map with portalCache in this pr. For optimize the code about the `_sqlTransformer`, we may handle it in another ticket.